src/App.js

```diff
@@ -13,37 +13,57 @@ import { contextApi } from "./contextApi";
 function App() {
   const [products, setProducts] = useState([]);
   const [userData, setUserData] = useState(false);
+  const [loading, setLoading] = useState(false);
   useEffect(() => {
-    axios.get("http://localhost:3001/allProducts").then((res) => {
-      setProducts(res.data);
-    });
+    setLoading(true);
+    axios
+      .get("http://localhost:3001/allProducts")
+      .then((res) => {
+        setProducts(res.data);
+      })
+      .catch((error) => {
+        console.log(error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
   }, []);
   const userDisconnect = () => {
     setUserData(false);
   };
   const userConnect = (user) => {
     setUserData(user);
   };
+  const addToCart = (valProduct) => {
+    userData.cart.unshift(valProduct);
+    setUserData([...userData]);
+  }
+  const deleteFromCart = (indexProduct) => {
+    userData.cart.splice(indexProduct, 1);
```

Comment:
This name is not clear, is `indexproduct` an object, maybe a number? consider change this into better name